modin/core/execution/unidist/implementations/pandas_on_unidist/io/io.py

```diff
@@ -0,0 +1,323 @@
+            """
+            partition_idx = kw["partition_idx"]
+            # the copy is made to not implicitly change the input parameters;
+            # to write to an intermediate buffer, we need to change `path_or_buf` in kwargs
+            csv_kwargs = kwargs.copy()
+            if partition_idx != 0:
+                # we need to create a new file only for first recording
+                # all the rest should be recorded in appending mode
+                if "w" in csv_kwargs["mode"]:
+                    csv_kwargs["mode"] = csv_kwargs["mode"].replace("w", "a")
+                # It is enough to write the header for the first partition
+                csv_kwargs["header"] = False
+
+            # for parallelization purposes, each partition is written to an intermediate buffer
+            path_or_buf = csv_kwargs["path_or_buf"]
+            is_binary = "b" in csv_kwargs["mode"]
+            csv_kwargs["path_or_buf"] = io.BytesIO() if is_binary else io.StringIO()
+            df.to_csv(**csv_kwargs)
+            content = csv_kwargs["path_or_buf"].getvalue()
+            csv_kwargs["path_or_buf"].close()
+
+            # each process waits for its turn to write to a file
+            UnidistWrapper.materialize(signals.wait.remote(partition_idx))
+
+            # preparing to write data from the buffer to a file
+            with pd_get_handle(
+                path_or_buf,
+                # in case when using URL in implicit text mode
+                # pandas try to open `path_or_buf` in binary mode
+                csv_kwargs["mode"] if is_binary else csv_kwargs["mode"] + "t",
+                encoding=kwargs["encoding"],
+                errors=kwargs["errors"],
+                compression=kwargs["compression"],
+                storage_options=kwargs.get("storage_options", None),
+                is_text=not is_binary,
+            ) as handles:
+                handles.handle.write(content)
+
+            # signal that the next process can start writing to the file
+            UnidistWrapper.materialize(signals.send.remote(partition_idx + 1))
+            # used for synchronization purposes
+            return pandas.DataFrame()
+
+        # signaling that the partition with id==0 can be written to the file
+        UnidistWrapper.materialize(signals.send.remote(0))
+        # Ensure that the metadata is syncrhonized
+        qc._modin_frame._propagate_index_objs(axis=None)
+        result = qc._modin_frame._partition_mgr_cls.map_axis_partitions(
+            axis=1,
+            partitions=qc._modin_frame._partitions,
+            map_func=func,
+            keep_partitioning=True,
+            lengths=None,
+            enumerate_partitions=True,
+            max_retries=0,
+        )
+        # pending completion
+        UnidistWrapper.materialize(
+            [partition.list_of_blocks[0] for partition in result.flatten()]
+        )
+
+    @staticmethod
+    def _to_parquet_check_support(kwargs):
```

Comment:
```suggestion
    _to_parquet_check_support = PandasOnRayIO._to_parquet_check_support 
```

Reply:
I do not want unidist to depend on other Modin engines that may require their installation. So I prefer to leave as is.